File: win-prog/ProgWinVB/Taming the Mouse/MouseConnectWaitCursor/MouseConnectWaitCursor.vb
' ----------------------------------------------------
' MouseConnectWaitCursor.vb (c) 2002 by Charles Petzold
' ----------------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class MouseConnectWaitCursor
    Inherits MouseConnect

    Shared Shadows Sub Main()
        Application.Run(New MouseConnectWaitCursor())
    End Sub

    Sub New()
        Text = "Mouse Connect with Wait Cursor"
    End Sub

    Protected Overrides Sub OnPaint(ByVal pea As PaintEventArgs)
        Cursor.Current = Cursors.WaitCursor
        Cursor.Show()
        MyBase.OnPaint(pea)
        Cursor.Hide()
        Cursor.Current = Cursors.Arrow
    End Sub
End Class

File: win-prog/ProgWinCS/Taming the Mouse/MouseConnectWaitCursor/MouseConnectWaitCursor.cs
// ----------------------------------------------------
// MouseConnectWaitCursor.cs © 2001 by Charles Petzold
// ----------------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class MouseConnectWaitCursor: MouseConnect
{
     public new static void Main()
     {
          Application.Run(new MouseConnectWaitCursor());
     }
     public MouseConnectWaitCursor()
     {
          Text = "Mouse Connect with Wait Cursor";
     }
     protected override void OnPaint(PaintEventArgs pea)
     {
          Cursor.Current = Cursors.WaitCursor;
          Cursor.Show();

          base.OnPaint(pea);

          Cursor.Hide();
          Cursor.Current = Cursors.Arrow;
     }
}
